front/pages/api/w/[wId]/labs/transcripts.ts

```diff
@@ -0,0 +1,153 @@
+import type { WithAPIErrorReponse } from "@dust-tt/types";
+import type { NextApiRequest, NextApiResponse } from "next";
+
+import { Authenticator, getSession } from "@app/lib/auth";
+import type { LabsTranscriptsProviderType } from "@app/lib/labs/transcripts/utils/types";
+import { LabsTranscriptsConfigurationResource } from "@app/lib/resources/labs_transcripts_configuration_resource";
+import { apiError, withLogging } from "@app/logger/withlogging";
+import { launchRetrieveNewTranscriptsWorkflow } from "@app/temporal/labs/client";
+
+export type GetLabsTranscriptsConfigurationResponseBody = {
+  configuration: LabsTranscriptsConfigurationResource | null;
+};
+
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<
+    WithAPIErrorReponse<GetLabsTranscriptsConfigurationResponseBody>
+  >
+): Promise<void> {
+  const session = await getSession(req, res);
+  const auth = await Authenticator.fromSession(
+    session,
+    req.query.wId as string
+  );
+
+  const owner = auth.workspace();
+  if (!owner) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "workspace_not_found",
+        message: "The workspace was not found.",
+      },
+    });
+  }
+
+  switch (req.method) {
+    case "GET":
+      const transcriptsConfigurationGetRes =
+        await LabsTranscriptsConfigurationResource.findByUserIdAndProvider({
+          userId: owner.id,
+          provider: req.query.provider as LabsTranscriptsProviderType,
+        });
+
+      return res
+        .status(200)
+        .json({ configuration: transcriptsConfigurationGetRes });
+
+    // Update
+    case "PATCH":
+      const {
```

Comment:
You need to use io-ts to define the expected body.

Reply:
Example:

https://github.com/dust-tt/dust/blob/1cc17f021c4858416c4085c3ce55841810100c9a/front/pages/api/w/%5BwId%5D/data_sources/managed.ts#L36-L45